fms/testing/_internal/hf/model_test_suite.py

```diff
@@ -163,6 +181,36 @@ def test_hf_model_compile_no_graph_breaks(self, fms_hf_model):
             pytest.fail(f"Failed to get signature of full-graph compiled model:\n{e}")
 
 
+class HFAutoModelTestSuite(HFModelFixtureMixin):
+    def test_hf_automodel_headless(self, fms_hf_model: PreTrainedModel):
+        """test that the headless model can be loaded with automodel after registration"""
+        register_fms_models()
```

Comment:
somewhat unrelated to this specific PR -- can this happen when the model(s) are imported rather than having to call this every time an HF-adapted model is used? e.g. something like in https://github.com/foundation-model-stack/foundation-model-stack/pull/57

Reply:
ideally register_fms_models is called once. If a user does have some import of the model, then this can be called there as well. In most cases, a user does not have an import of the model as they are using the function so that they can use AutoModels with a path in a script without importing. I currently am implicitly calling this here as well (https://github.com/foundation-model-stack/foundation-model-stack/blob/82cdd60162d8d2dc5822ace90610e95c94d51b90/fms/models/hf/llama/__init__.py#L26) that way a user can just say get_model and it will automatically register (this will be superseded by the get_model api i would assume)